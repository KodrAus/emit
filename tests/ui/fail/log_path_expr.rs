#![feature(stmt_expr_attributes, proc_macro_hygiene)]

struct Data {
    a: i32,
}

fn main() {
    tracing_subscriber::fmt().init();

    let data = Data {
        a: 42,
    };

    emit::info!("Logging with a deeply nested {data.a}");
}
