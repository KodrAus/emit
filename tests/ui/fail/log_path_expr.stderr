error: proc macro panicked
  --> fail/log_path_expr.rs:14:5
   |
14 |     emit::info!("Logging with a deeply nested {data.a}");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: message: failed to parse: Error { reason: "failed to parse `data.a` as an expression", source: Some(Error("unexpected token")) }
